Packages should be using peerDependencies

For a package.json with the following:

```json
"devDependendencies": {
    "eslint": "<some_version>",
    "@ni/eslint-config-angular": "<some_version>"
}
```

It is reasonable to expect that dependency resolution will work correctly and all the related dependencies (`@ni/eslint-config-typescript`, etc) will be installed.

However, because [`@ni/eslint-config-angular` lists `@ni/eslint-config-typescript` as a dependency](https://github.com/ni/javascript-styleguide/blob/fb056718f8e5d703797e83a3f9b0fe235b39aceb/packages/eslint-config-angular/package.json#L34) instead of a peerDependency, npm is allowed to resolve the package structure as:

- node_modules
  - `@ni/eslint-config-angular` 
    - node_modules
      - `@ni/eslint-config-typescript
      - `@typescript-eslint/eslint-plugin`
      - `<other typescript eslint plugins>`

Which results in error from the `eslint` because packages expected to be at the root are not:
```
[error] Error: Failed to load plugin '@typescript-eslint' declared in '.eslintrc.json#overrides[0] » @ni/eslint-config-angular » @ni/eslint-config-typescript': Cannot find module '@typescript-eslint/eslint-plugin'
```

The expected module resolution tree should look like:

- node_modules
  - `@ni/eslint-config-angular` 
  - `@ni/eslint-config-typescript`
  - `@typescript-eslint/eslint-plugin`
  - `<all other eslint config plugins>`

To achieve that consistently the packages should all move to peer dependencies, ie:

https://github.com/ni/javascript-styleguide/blob/fb056718f8e5d703797e83a3f9b0fe235b39aceb/packages/eslint-config-angular/package.json#L33-L35
https://github.com/ni/javascript-styleguide/blob/fb056718f8e5d703797e83a3f9b0fe235b39aceb/packages/eslint-config-typescript/package.json#L34-L36
https://github.com/ni/javascript-styleguide/blob/fb056718f8e5d703797e83a3f9b0fe235b39aceb/packages/eslint-config-javascript/package.json#L32-L34

As a workaround you can push npm to resolve packages into the root by adding wildcard dependencies for the sub packages until it starts working, i.e.:

```json
"devDependendencies": {
    "eslint": "<some_version>",
    "@ni/eslint-config-angular": "<some_version>",
    "@ni/eslint-config-typescript": "*"
}
```

--- a/change/@ni-eslint-config-angular-1b5b8d6f-05be-4e05-8bdd-dacc3b74b73e.json
+++ b/change/@ni-eslint-config-angular-1b5b8d6f-05be-4e05-8bdd-dacc3b74b73e.json
@@ -1,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "Make all eslint config deps peers",
+  "packageName": "@ni/eslint-config-angular",
+  "email": "rajsite@users.noreply.github.com",
+  "dependentChangeType": "patch"
+}

--- a/change/@ni-eslint-config-javascript-75b2f17b-c1af-46a6-a28a-cbc8728816f1.json
+++ b/change/@ni-eslint-config-javascript-75b2f17b-c1af-46a6-a28a-cbc8728816f1.json
@@ -1,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "Make all eslint config deps peers",
+  "packageName": "@ni/eslint-config-javascript",
+  "email": "rajsite@users.noreply.github.com",
+  "dependentChangeType": "patch"
+}

--- a/change/@ni-eslint-config-typescript-8e083390-e1d2-4aef-a76f-d722b85ca61e.json
+++ b/change/@ni-eslint-config-typescript-8e083390-e1d2-4aef-a76f-d722b85ca61e.json
@@ -1,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "Make all eslint config deps peers",
+  "packageName": "@ni/eslint-config-typescript",
+  "email": "rajsite@users.noreply.github.com",
+  "dependentChangeType": "patch"
+}

--- a/packages/eslint-config-angular/package.json
+++ b/packages/eslint-config-angular/package.json
@@ -30,14 +30,12 @@
     "/*.js",
     "!/.*.js"
   ],
-  "dependencies": {
-    "@ni/eslint-config-typescript": "^4.2.0"
-  },
   "peerDependencies": {
     "@angular-eslint/builder": "^14.0.2",
     "@angular-eslint/eslint-plugin": "^14.0.2",
     "@angular-eslint/eslint-plugin-template": "^14.0.2",
     "@angular-eslint/schematics": "^14.0.2",
-    "@angular-eslint/template-parser": "^14.0.2"
+    "@angular-eslint/template-parser": "^14.0.2",
+    "@ni/eslint-config-typescript": "^4.2.0"
   }
 }

--- a/packages/eslint-config-javascript/package.json
+++ b/packages/eslint-config-javascript/package.json
@@ -29,11 +29,9 @@
     "/*.js",
     "!/.*.js"
   ],
-  "dependencies": {
-    "eslint-config-airbnb-base": "~15.0.0"
-  },
   "peerDependencies": {
     "eslint": "^8.23.0",
+    "eslint-config-airbnb-base": "~15.0.0",
     "eslint-plugin-import": "^2.26.0"
   }
 }

--- a/packages/eslint-config-typescript/package.json
+++ b/packages/eslint-config-typescript/package.json
@@ -31,10 +31,8 @@
     "!/.*.js",
     "/lib/*"
   ],
-  "dependencies": {
-    "@ni/eslint-config-javascript": "^4.2.0"
-  },
   "peerDependencies": {
+    "@ni/eslint-config-javascript": "^4.2.0",
     "@typescript-eslint/parser": "^5.36.1",
     "@typescript-eslint/eslint-plugin": "^5.36.1"
   }

